static/app/utils/withCustomerDomain.tsx

```diff
@@ -0,0 +1,92 @@
+import {useContext} from 'react';
+import {RouteComponent, RouteComponentProps} from 'react-router';
+import trimEnd from 'lodash/trimEnd';
+import trimStart from 'lodash/trimStart';
+
+import localStorage from 'sentry/utils/localStorage';
+import recreateRoute from 'sentry/utils/recreateRoute';
+import replaceRouterParams from 'sentry/utils/replaceRouterParams';
+import {OrganizationContext} from 'sentry/views/organizationContext';
+
+type WithCustomerDomainOptions = {
+  redirect: Array<{from: string; to: string}>;
+};
+
+function withCustomerDomain<P extends RouteComponentProps<{}, {}>>(
```

Comment:
Would be good to have a description of what this is doing/why for folks outside of hybrid cloud.